lib/buffer.js

```diff
@@ -320,6 +319,42 @@ function allocate(size) {
 }
 
 
+// Ported from deps/v8/src/unicode-inl.h
+function isTrailSurrogate(code) {
+  return (code & 0xfc00) === 0xdc00;
+}
+
+
+function isLeadSurrogate(code) {
+  // No previous character
+  if (code === -1) return false;
+  return (code & 0xfc00) === 0xd800;
+}
+
+
+function byteLengthUtf8(string) {
+  var len = 0;
+  var previous = -1;
+  for (var i = 0; i < string.length; i++) {
+    // NOTE: 0 <= code <= 0xffff
+    var code = string.charCodeAt(i);
+    if (code <= 0x7f) {
+      len++;
```

Comment:
`len += 1;` for consistency with line 348.

Reply:
Nitpicking :wink: